File: sdk/rust/tests/drafts_test.rs
use agentmailer::prelude::*;

mod wire_test_utils;

#[tokio::test]
#[allow(unused_variables, unreachable_code)]
async fn test_drafts_list_with_wiremock() {
    wire_test_utils::reset_wiremock_requests().await.unwrap();
    let wiremock_base_url = wire_test_utils::get_wiremock_base_url();

    let mut config = ClientConfig {
        token: Some("<token>".to_string()),
        ..Default::default()
    };
    config.base_url = wiremock_base_url.to_string();
    let client = AgentMailer::new(config).expect("Failed to build client");

    let result = client.drafts.list(&"inboxId".to_string(), None).await;

    assert!(result.is_ok(), "Client method call should succeed");

    wire_test_utils::verify_request_count("GET", "/v1/inboxes/inboxId/drafts", None, 1)
        .await
        .unwrap();
}

#[tokio::test]
#[allow(unused_variables, unreachable_code)]
async fn test_drafts_create_with_wiremock() {
    wire_test_utils::reset_wiremock_requests().await.unwrap();
    let wiremock_base_url = wire_test_utils::get_wiremock_base_url();

    let mut config = ClientConfig {
        token: Some("<token>".to_string()),
        ..Default::default()
    };
    config.base_url = wiremock_base_url.to_string();
    let client = AgentMailer::new(config).expect("Failed to build client");

    let result = client
        .drafts
        .create(
            &"inboxId".to_string(),
            &DraftCreate {
                ..Default::default()
            },
            None,
        )
        .await;

    assert!(result.is_ok(), "Client method call should succeed");

    wire_test_utils::verify_request_count("POST", "/v1/inboxes/inboxId/drafts", None, 1)
        .await
        .unwrap();
}

#[tokio::test]
#[allow(unused_variables, unreachable_code)]
async fn test_drafts_get_with_wiremock() {
    wire_test_utils::reset_wiremock_requests().await.unwrap();
    let wiremock_base_url = wire_test_utils::get_wiremock_base_url();

    let mut config = ClientConfig {
        token: Some("<token>".to_string()),
        ..Default::default()
    };
    config.base_url = wiremock_base_url.to_string();
    let client = AgentMailer::new(config).expect("Failed to build client");

    let result = client
        .drafts
        .get(&"inboxId".to_string(), &"draftId".to_string(), None)
        .await;

    assert!(result.is_ok(), "Client method call should succeed");

    wire_test_utils::verify_request_count("GET", "/v1/inboxes/inboxId/drafts/draftId", None, 1)
        .await
        .unwrap();
}

#[tokio::test]
#[allow(unused_variables, unreachable_code)]
async fn test_drafts_delete_with_wiremock() {
    wire_test_utils::reset_wiremock_requests().await.unwrap();
    let wiremock_base_url = wire_test_utils::get_wiremock_base_url();

    let mut config = ClientConfig {
        token: Some("<token>".to_string()),
        ..Default::default()
    };
    config.base_url = wiremock_base_url.to_string();
    let client = AgentMailer::new(config).expect("Failed to build client");

    let result = client
        .drafts
        .delete(&"inboxId".to_string(), &"draftId".to_string(), None)
        .await;

    assert!(result.is_ok(), "Client method call should succeed");

    wire_test_utils::verify_request_count("DELETE", "/v1/inboxes/inboxId/drafts/draftId", None, 1)
        .await
        .unwrap();
}

#[tokio::test]
#[allow(unused_variables, unreachable_code)]
async fn test_drafts_update_with_wiremock() {
    wire_test_utils::reset_wiremock_requests().await.unwrap();
    let wiremock_base_url = wire_test_utils::get_wiremock_base_url();

    let mut config = ClientConfig {
        token: Some("<token>".to_string()),
        ..Default::default()
    };
    config.base_url = wiremock_base_url.to_string();
    let client = AgentMailer::new(config).expect("Failed to build client");

    let result = client
        .drafts
        .update(
            &"inboxId".to_string(),
            &"draftId".to_string(),
            &DraftUpdate {
                ..Default::default()
            },
            None,
        )
        .await;

    assert!(result.is_ok(), "Client method call should succeed");

    wire_test_utils::verify_request_count("PATCH", "/v1/inboxes/inboxId/drafts/draftId", None, 1)
        .await
        .unwrap();
}

#[tokio::test]
#[allow(unused_variables, unreachable_code)]
async fn test_drafts_send_with_wiremock() {
    wire_test_utils::reset_wiremock_requests().await.unwrap();
    let wiremock_base_url = wire_test_utils::get_wiremock_base_url();

    let mut config = ClientConfig {
        token: Some("<token>".to_string()),
        ..Default::default()
    };
    config.base_url = wiremock_base_url.to_string();
    let client = AgentMailer::new(config).expect("Failed to build client");

    let result = client
        .drafts
        .send(&"inboxId".to_string(), &"draftId".to_string(), None)
        .await;

    assert!(result.is_ok(), "Client method call should succeed");

    wire_test_utils::verify_request_count(
        "POST",
        "/v1/inboxes/inboxId/drafts/draftId/send",
        None,
        1,
    )
    .await
    .unwrap();
}

File: sdk/ruby/test/wire/drafts_test.rb
# frozen_string_literal: true

require_relative "wiremock_test_case"

class DraftsWireTest < WireMockTestCase
  def setup
    super

    @client = AgentMailer::Client.new(
      api_key: "<token>",
      base_url: WIREMOCK_BASE_URL
    )
  end

  def test_drafts_list_with_wiremock
    test_id = "drafts.list.0"

    @client.drafts.list(
      inbox_id: "inboxId",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "drafts.list.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "GET",
      url_path: "/v1/inboxes/inboxId/drafts",
      query_params: nil,
      expected: 1
    )
  end

  def test_drafts_create_with_wiremock
    test_id = "drafts.create.0"

    @client.drafts.create(
      inbox_id: "inboxId",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "drafts.create.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "POST",
      url_path: "/v1/inboxes/inboxId/drafts",
      query_params: nil,
      expected: 1
    )
  end

  def test_drafts_get_with_wiremock
    test_id = "drafts.get.0"

    @client.drafts.get(
      inbox_id: "inboxId",
      draft_id: "draftId",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "drafts.get.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "GET",
      url_path: "/v1/inboxes/inboxId/drafts/draftId",
      query_params: nil,
      expected: 1
    )
  end

  def test_drafts_delete_with_wiremock
    test_id = "drafts.delete.0"

    @client.drafts.delete(
      inbox_id: "inboxId",
      draft_id: "draftId",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "drafts.delete.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "DELETE",
      url_path: "/v1/inboxes/inboxId/drafts/draftId",
      query_params: nil,
      expected: 1
    )
  end

  def test_drafts_update_with_wiremock
    test_id = "drafts.update.0"

    @client.drafts.update(
      inbox_id: "inboxId",
      draft_id: "draftId",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "drafts.update.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "PATCH",
      url_path: "/v1/inboxes/inboxId/drafts/draftId",
      query_params: nil,
      expected: 1
    )
  end

  def test_drafts_send__with_wiremock
    test_id = "drafts.send_.0"

    @client.drafts.send_(
      inbox_id: "inboxId",
      draft_id: "draftId",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "drafts.send_.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "POST",
      url_path: "/v1/inboxes/inboxId/drafts/draftId/send",
      query_params: nil,
      expected: 1
    )
  end
end
